src/iccpd/src/mlacp_sync_update.c

```diff
@@ -0,0 +1,659 @@
+                        mac_msg->age_flag = set_mac_local_age_flag(csm, mac_msg, 1);
+
+                        /*Send mac add message to mclagsyncd*/
+                        add_mac_to_chip(mac_msg, MAC_TYPE_DYNAMIC);
+                    }
+                }
+
+                #if 0
+                mac_msg->op_type = MAC_SYNC_ACK;
+                if (iccp_csm_init_msg(&msg_send, (char*)mac_msg, sizeof(struct MACMsg)) == 0)
+                {
+                    /*Reply mac ack message to peer, peer will clean MAC_AGE_PEER flag*/
+                    TAILQ_INSERT_TAIL(&(MLACP(csm).mac_msg_list), msg_send, tail);
+                    ICCPD_LOG_DEBUG(__FUNCTION__, "Recv ADD, MAC-msg-list enqueue: %s, add %s vlan-id %d, op_type %d",
+                                    mac_msg->ifname, mac_msg->mac_str, mac_msg->vid, mac_msg->op_type);
+                }
+                #endif
+            }
+            else if (tlv->type == MAC_SYNC_ACK)
+            {
+                /*Clean the MAC_AGE_PEER flag*/
+                mac_msg->age_flag &= ~MAC_AGE_PEER;
+                ICCPD_LOG_DEBUG(__FUNCTION__, "Recv ACK, Remove peer age flag:%d ifname  %s, add %s vlan-id %d, op_type %d",
+                                mac_msg->age_flag, mac_msg->ifname, mac_msg->mac_str, mac_msg->vid, mac_msg->op_type);
+            }
+
+            break;
+        }
+    }
+
+    /* delete/add MAC list*/
+    if (msg && tlv->type == MAC_SYNC_DEL)
+    {
+        mac_msg->age_flag |= MAC_AGE_PEER;
+        ICCPD_LOG_DEBUG(__FUNCTION__, "Add peer age flag: %d   ifname %s, add %s vlan-id %d, op_type %d",
+                        mac_msg->age_flag, mac_msg->ifname, mac_msg->mac_str, mac_msg->vid, mac_msg->op_type);
+
+        if (mac_msg->age_flag == (MAC_AGE_LOCAL | MAC_AGE_PEER))
+        {
+            /*send mac del message to mclagsyncd.*/
+            del_mac_from_chip(mac_msg);
+
+            /*If local and peer both aged, del the mac*/
+            TAILQ_REMOVE(&(MLACP(csm).mac_list), msg, tail);
+            free(msg->buf);
+            free(msg);
+        }
+        else
+        {
+            return 0;
+        }
+    }
+    else if (!msg && tlv->type == MAC_SYNC_ADD)
+    {
+        mac_msg = (struct MACMsg*)&mac_data;
+        mac_msg->fdb_type = MAC_TYPE_DYNAMIC;
+        mac_msg->vid = ntohs(tlv->vid);
+        sprintf(mac_msg->mac_str, "%s", tlv->mac_str);
+        sprintf(mac_msg->ifname, "%s", tlv->ifname);
+        sprintf(mac_msg->origin_ifname, "%s", tlv->ifname);
+        mac_msg->age_flag = 0;
+
+        /*If the MAC is learned from orphan port, or from MCLAG port but the local port is down*/
+        if (from_mclag_intf == 0 || local_if->state == PORT_STATE_DOWN)
+        {
+            /*Redirect the mac to peer-link*/
+            memcpy(&mac_msg->ifname, csm->peer_itf_name, IFNAMSIZ);
+
+            /*Set MAC_AGE_LOCAL flag*/
+            mac_msg->age_flag = set_mac_local_age_flag(csm, mac_msg, 1);
```

Comment:
Same as one of the earlier comments added at line 226  ,calling set_mac_local_age_flag from this place can cause unnecessary updates while learning remote mac for first time.

Reply:
The MAC will be deleted only when both of flags are set. Why this is unnecessary? If you have better idea to deal with MAC sync and age, please share to us. Thanks.